lib/mapObjects/CGPandoraBox.cpp

```diff
@@ -25,297 +25,119 @@
 
 VCMI_LIB_NAMESPACE_BEGIN
 
-void CGPandoraBox::initObj(CRandomGenerator & rand)
+void CGPandoraBox::init()
 {
-	blockVisit = (ID==Obj::PANDORAS_BOX); //block only if it's really pandora's box (events also derive from that class)
-	hasGuardians = stacks.size();
+	blockVisit = true;
+	configuration.selectMode = Rewardable::SELECT_ALL;
```

Comment:
I assume only pandora's from VCMI map can have multiple rewards?

Reply:
no, just H3 different reward types are in separate tabs, you need to select different dropdown item to pick different reward type, but what you chose in another tab is preserved

Reply:
My question is basically why pandoras use `Rewardable::SELECT_ALL` mode instead of using SELECT_FIRST like all other objects with single reward.

My guess is that in some cases Pandora's might have multiple rewards, but in H3 there is only one (but potentially quite large one). So can't think of anything else other than VCMI map

Reply:
Pandora always could have multiple rewards. It has a single reward only on random maps.

Reply:
You mean like mana and morale?
Those are parts of single "reward" as far as our rewardable object system is concerned.

Reply:
Ahh, alright.

Reply:
It's more implementation-vise. For pandoras different rewards are shown as different messages. So I try to preserve this OH3 behavior.

Reply:
> For pandoras different rewards are shown as different messages

Are you sure?

![зображення](https://github.com/vcmi/vcmi/assets/1576820/adf83ade-6e47-4762-b1df-90d39a9b9dd0)

Reply:
And also this may be related: https://github.com/vcmi/vcmi/issues/2514
(not pandora, but event, however they are largely identical)

Reply:
Try to add creatures into pandora box

Reply:
Looks like H3 has automated splitting of message by type:
- Experience, mana, luck, morale, primary skills, secondary skills go into single window but ONLY if total number of components is below 8
- Artifacts, creatures, resources always go into separate window.

Also, I think we already have message splitting on client - made by rilian-la-te as part of info box notifications and Scholar overflow fix.

So not sure if we need to split pandora reward on server side.